### ISSUE TITLE
Conflicting EXE Files, can this be defined?

### ISSUE THREAD

**Issue by ssgoku129** [phin05/discord-rich-presence-plex#30]
I have Discord running under a few profiles or multiple discords running under 1 profile, how does this script prioritize which Discord.exe gets chosen to send the rich presence to?

I.e. if 2 discord clients are running on the same profile - Alpha and regular, can I make it send the rich presence to the alpha client?

Furthermore, if 2 admin profiles have access to the same computer and both have Discord installed under their profile, it seems that running the python script under 1 profile will use the 2nd profile's Discord client if it was started before the 1st profile started Discord.

Basically it will just take the first Discord process it sees and apply rich presence there.

Can I define either the EXE or ID for Discord user for it to send the rich presence to?

**Comment by phin05** [comment]
> Can I define either the EXE or ID for Discord user for it to send the rich presence to?

It's only possible to specify the Discord IPC pipe number. Discord's IPC pipe name can end with a number in the range of `0-9`. When a Discord client is launched, it binds to the first unbound pipe number, which is typically `0`. Only `0` was being used by the script earlier, but now by default it will go through all the possible pipe numbers and use the first existing one. You can configure this pipe number starting from [v2.4.2](https://github.com/phin05/discord-rich-presence-plex/releases/tag/v2.4.2). Refer to the [configuration reference](https://github.com/phin05/discord-rich-presence-plex/blob/master/README.md#reference) (user -> server -> ipcPipeNumber) in the README.
